platform-sdk/docs/proposals/DS-0x/DS - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,458 @@
+
+### External References
+- https://andrea.corbellini.name/2015/05/17/elliptic-curve-cryptography-a-gentle-introduction/
+- https://www.iacr.org/archive/asiacrypt2001/22480516.pdf
+- https://hackmd.io/@benjaminion/bls12-381#Motivation
+- https://www.johannes-bauer.com/compsci/ecc/
+
+## Changes
+
+### Core Behaviors
+
+The proposed TSS solution combines EC-Cryptography primitives that allow for short aggregable signatures, assignation of multiple shares per shareholder to represent participants' weight,
+and combines Shamir's secret sharing with ElGamal public key encryption and groth21 for zero-knowledge proofs.
+
+#### Description of the Scheme
+
+An operator will generate a long-living persisting ec private/public key pair for each participant of the scheme.
+The operator will distribute the public key of every participant to the other participants in the scheme.
+
+The following section describes the process from a particular participant's point of view after the key distribution happened:
+
+##### Input
+* Participant's lifelong persisting EC private key (Private to each participant)
+* Number of participants (Public)
+* Amount of shares per participant (Public)
+* All participants' lifelong persisting  EC public key (Public)
+* A predefined `SignatureSchema` that defines the type of Curve to use, and which Group of the Pairing is used for PublicKey Generation (Public)
+
+#####  Bootstrap Stage
+
+
+(Shamir’s Secret Sharing)
+###### 1. Create  shares
+Given directory that identifies each participant uniquely, their amount of shares and EC Public Keys; the current participant and its EC Private Key, and the predefined threshold
+
+E.g: In a scheme distributing 10 shares over 4 participants
+```
+Directory:
+P   # shares
+-----------------------------
+P₁  5        P₁_EC_PublicKey
+P₂  2        P₂_EC_PublicKey
+P₃  1        P₃_EC_PublicKey
+P₄  2        P₄_EC_PublicKey
+
+Current Participant: (P₁;  P₁_EC_PrivateKey)
+threshold: 5
+```
+
+###### 2. Create ShamSecretShareMessage 
+(Shamir's secret-sharing: polynomial interpolation)
+
+Create `ShareId` directory
+```
+sid₁	sid₂	sid₃	sid₄	sid₅	sid₆	sid₇	sid₈	sid₉	sid₁₀
+P₁  	P₁  	P₁  	P₁  	P₁  	P₂  	P₂  	P₃  	P₄  	P₄  
+```
+
+Given participant's `privateKey`: `k` (the secret being shared), each shareholder will produce `n` (n=total number of shares) values `xₛ` by introducing `k`'s EC field element in a polynomial `Pₖ` for each `ShareId`: `sidₛ` in the directory
+
+```
+   xₛ=Pₖ(sidₛ) for privateKey: k and each ShareId: sidₛ
+```
+Each `xₛ` constitutes points in the polynomial.
+
+The polynomial `Pₖ` is a polynomial with grade `t-1` (t=threshold) created with random coefficients from `SignatureScheme.publicKeyGroup` and `k`'s EC field element in `aₒ` and `x` each `sidₛ`
+
+(ElGammal)
+Once each `xₛ` value has been calculated for each `ShareId`: `sidₛ`, the value: `SecretShareMessage`:`Mₛ` will be produced by encrypting `xₖ` using `sidₛ` owner's publicKey.
```

Comment:
> Once each `xₛ` value has been calculated for each `ShareId`: `sidₛ`...

Nit: there is only 1 `xₛ` for each share ID, so the sentence should read

"Once _the_ `xₛ` value has been calculated for each `ShareId`: `sidₛ`"...